spec/components/application_complete_content_component_spec.rb

```diff
@@ -19,7 +19,7 @@ def render_result
   end
 
   it 'renders with correct respond by date' do
-    expect(render_result.text).to include('1 December 2019')
+    expect(render_result.text).to include('17 December 2019')
```

Comment:
It's a bit of a "smell" that we're having to change this spec as a result of changes in the application form business logic. In this spec, we are coupling an actual factory-created application form object into the component, meaning that any changes to the application form logic class breaks the spec.

One proposal would be to make  `application_form` in `render_result` a stub, which stubs out the application form methods that are called by the component. This would also make it much more explicit where the dates (e.g. 17 December 2019) are coming from.

Reply:
@benilovj I've stubbed out the `ApplicationForm` and more importantly `ApplicationDates` so the component test should now have no external implementation dependencies. Let me know if this makes it cleaner or not.

Reply:
Yes, this feels much clearer now, thanks.